File: rust/383.ransom-note.rs
/*
 * @lc app=leetcode id=383 lang=rust
 *
 * [383] Ransom Note
 */

// @lc code=start
impl Solution {
    pub fn can_construct(ransom_note: String, magazine: String) -> bool {
        if ransom_note.len() > magazine.len() {
            return false;
        }
        let mut chars = vec![0; 26];
        for c in ransom_note.bytes() {
            chars[(c - b'a') as usize] += 1;
        }
        for c in magazine.bytes() {
            chars[(c - b'a') as usize] -= 1;
        }
        return !chars.iter().any(|x| *x > 0);
    }
}
// @lc code=end

File: javascript/383.ransom-note.js
/*
 * @lc app=leetcode id=383 lang=javascript
 *
 * [383] Ransom Note
 */

// @lc code=start
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function (ransomNote, magazine) {
  if (ransomNote.length > magazine.length) return false;

  const chars = new Array(26).fill(0);
  const d = "a".charCodeAt();
  for (const c of ransomNote) {
    chars[c.charCodeAt() - d]++;
  }
  for (const c of magazine) {
    chars[c.charCodeAt() - d]--;
  }
  return !chars.some((n) => n > 0);
};
// @lc code=end
